PermanentOrg/web-app
# Fetch featured archives from API

This PR updates the public gallery to fetch the Featured Archives from Stela's `/archive/featured` endpoint instead of fetching them from a hardcoded value in production or an environment variable in lower environments. Some of this work was done during Permanent's staff retreat in Albuquerque!

This PR experiments with some new (to this codebase) software architecture patterns; instead of putting the code to fetch featured archives in the big centralized `ApiService`, the `GalleryModule` owns the code. This is because only this module will ever have to fetch Featured Archives, so it doesn't make sense for every other part of the codebase that imports the `ApiModule` to suddenly have access to this new function.

I've also experimented with some form of Interface Segregation, so the GalleryComponent only knows of this new API function through an interface. This was originally because I was planning to hook it up in such a way that the GalleryComponent would still be using the ApiService, but only through the FeaturedArchiveApi interface. That didn't end up working out as I had hoped, so the interface feels a bit more pointless... I would prefer to keep it though, since I think this is the way we want to handle API services in the future so that components only have access to the API methods they need. (Also, we can swap out the FeaturedArchiveService with something else later and the GalleryComponent will never know. This also makes mocking in tests actually type safe. I think both of these are neat.).

**Steps to Test:**
1. Set up featured archives on your local backend
2. Verify that the featured archives specified in the backend show up on the public gallery

**Deployment/Merging Notes**
This should not be merged and deployed until the production database is configured with Featured Archives.
In addition, once this is merged, we should delete the FEATURED_ARCHIVES secret used during the dev build.

Resolves PER-9375.
